pkg/provisioner/cloudprovisioner.go

```diff
@@ -340,7 +340,12 @@ func (c *CloudProvisioner) PublishVolume(
 			return nil, err
 		}
 
-		lun, err = c.cloud.AttachDisk(ctx, true, diskName, volumeID, nodeName, cachingMode, disk)
+		asyncAttach := true
```

Comment:
Should we update the driver-parameters.md file to specify  that the default value of this parameter for V2 driver is true?

Reply:
This PR is more or less so we can run the scale tests with synchronous `AttachDisk` to see its impact on the scale-out times. The full (and updated) change for this support in V1 will be brought in as part of #1216 .

Reply:
If this is just for testing, will this be reverted after #1216 is merged?

Reply:
We'll still need the updated `cloud-provisioner-azure` brought in by this change to enable sync/async operation via configuration to maintain feature parity with the v1 driver.

Reply:
I updated the V2 driver parameter documentation.